### PR TITLE
Create output file manifest

### LINKED ISSUE #3: also output manifest

### PR DESCRIPTION
**Pull request by pivnicek**
fixes #3

### CODE REVIEW

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 1]
```diff
@@ -34,13 +43,31 @@ private static void readConfigFile(String configFile) throws ApplicationExceptio
...
+        dbHost = db.getString("host");
+        dbUser = db.getString("user");
+        dbPassword = db.getString("#password");
+        dbName = db.getString("database");
+        query = params.getString("query");
+        outputFile = params.getString("outputFile");
+        outputTable = params.getString("outputTable");
+        if (params.has("incremental")) {
+            incremental = params.getBoolean("incremental");
+        }
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: else incremental = false ?

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 2]
```diff
@@ -71,31 +98,113 @@ private static void fetchData() throws UserException {
...
+
+    private static void writeManifest(
+        String outputFile,
+        ResultSetMetaData rsMeta,
+        DatabaseMetaData dbMeta
+    ) throws UserException, IOException {
+        try {
+            JSONArray columnNames = new JSONArray();
+            JSONArray tableMetadata = new JSONArray();
+            ArrayList<String> pkList = new ArrayList<String>();
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: ArrayList<String> pkList = new ArrayList<>();

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 3]
```diff
@@ -71,31 +98,113 @@ private static void fetchData() throws UserException {
...
+
+            for (int i = 1; i <= rsMeta.getColumnCount(); i++) {
+                // just get table metadata for the first column (this may result in bs results for complicated queries)
+                JSONArray columnMetadata = new JSONArray();
+                String name = rsMeta.getColumnName(i);
+                columnNames.put(name);
+                columnMetadata.put(makeMetadataObject("name", name));
+                columnMetadata.put(makeMetadataObject("ordinalPosition", String.valueOf(i)));
+                columnMetadata.put(makeMetadataObject("autoIncrement", String.valueOf(rsMeta.isAutoIncrement(i))));
+                columnMetadata.put(makeMetadataObject("primaryKey", String.valueOf(pkList.contains(name))));
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: are those mtd keys documented somewhere?

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 4]
```diff
@@ -71,31 +98,113 @@ private static void fetchData() throws UserException {
...
+            manifest.put("columnMetadata", allColumnMetadata);
+            manifest.put("destination", outputTable);
+            manifest.put("incremental", incremental);
+            if (primaryKey != null) {
+                manifest.put("primaryKey", primaryKey);
+            }
+            try (FileWriter file = new FileWriter(outputFile + ".manifest")) {
+                file.write(manifest.toString());
+                System.out.println("Successfully Copied JSON Object to File...");
+                System.out.println("\nJSON Object: " + manifest.toString());
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: del ?

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 5]
```diff
@@ -71,31 +98,113 @@ private static void fetchData() throws UserException {
...
+            }
+        } catch (SQLException e) {
+            throw new UserException("Error writing manifest: " + e.getMessage(), e);
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: this should be an app error?

File: src/main/java/com/keboola/tableexporter/Application.java [hunk 6]
```diff
@@ -34,13 +43,31 @@ private static void readConfigFile(String configFile) throws ApplicationExceptio
...
-        dbPassword = obj.getJSONObject("parameters").getJSONObject("db").getString("#password");
-        dbName = obj.getJSONObject("parameters").getJSONObject("db").getString("database");
-        query = obj.getJSONObject("parameters").getString("query");
-        outputFile = obj.getJSONObject("parameters").getString("outputFile");
+        JSONObject params = obj.getJSONObject("parameters");
+        JSONObject db = params.getJSONObject("db");
+        dbPort = db.getString("port");
+        dbHost = db.getString("host");
+        dbUser = db.getString("user");
+        dbPassword = db.getString("#password");
```
Comment (odinuv) [inline src/main/java/com/keboola/tableexporter/Application.java]: also i think we need some docs for the json file spec, it's getting non-trivial

**Comment by pivnicek** [comment]
thanks @odinuv !  yeah, I will put the mtd documentation at the top of my list.  maybe I'll put it in sites as a first step

File: src/main/java/com/keboola/tableexporter/Application.java (hunk 6 above)
Comment (pivnicek) [reply src/main/java/com/keboola/tableexporter/Application.java]: j, I'll put it in the readme

**Comment by pivnicek** [comment]
I jumped the gun a bit with this.  Although it's great that we can get metadata from all queries, still need to implement the column name sanitization here.

**Comment by pivnicek** [comment]
Also need to implement baseType mapping here

**Comment by pivnicek** [comment]
deprecated for the meantime

**Pull request closed without merging**